### PR TITLE
nice!nano board support

### LINKED ISSUE #1477: nice!nano board support
I'd like to add and use this board but I don't seem to have skills to do it myself.

It uses nRF52840, pinout: https://docs.nicekeyboards.com/#/nice!nano/pinout_schematic

I've been able to run blinky using `feather-nrf52840` target and changing the LED pin number: https://github.com/premek/tinygo-nicenano/commit/1506c56e4cab2a7e8f866b8c0d491fe77e49c282

I'm able to test with the board or if anyone can help I will try to make a PR, thanks!

### PR DESCRIPTION
**Pull request by premek**
Fixes #1477

I'm trying to add nicenano board support but having no experience in this I'd appreciate someone have a look at this.

Board documentation: https://docs.nicekeyboards.com/#/nice!nano/pinout_schematic

I copied Feather nRF52840 board as a starting point, renamed the pins based on the pinout. On the board the pins are labeled only with a number, e.g. "106" for P1_06, I named the constant D106. 

I need help with UART, I2C, SPI constants as I don't know much about that and cannot see anyting in the documentation and if I try to remove them `make` succeeds but compiling the blinky example fails. 

Also, in the JSON file I'm not sure about the following values:
- build-tags
- flash-1200-bps-reset
- uf2-family-id
- linkerscript

But in the current state this allows me to flash blinky1 example with the following command:
```
make && ./build/tinygo flash -target=nicenano src/examples/blinky1/blinky1.go
```

### CODE REVIEW

File: src/machine/board_nicenano.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+// USB CDC identifiers
+const (
+	usb_STRING_PRODUCT      = "nice!nano"
+	usb_STRING_MANUFACTURER = "Nice Keyboards"
+)
+
+var (
+	usb_VID uint16 = 0x239A
+	usb_PID uint16 = 0x0029
```
Comment (deadprogram) [inline src/machine/board_nicenano.go:83]: How did you determine this value should be used for the nicenano board?

**Comment by deadprogram** [comment]
Hi @premek thank you for your contribution.

Regarding the pin numbering, probably `D106` is about as good as `P1_06`, so seems alright.

Regarding the UART, I2C, and SPI, since the nRF52840 is very flexible on assigning IO, you could probably use whichever pins are broken out on the board and not used by anything else. The pinout/schematic does appear to show what is used/available https://github.com/Nicell/nice-keyboards-docs/blob/master/docs/nice!nano/pinout_schematic.md

The changes you made in the JSON appear to be correct.

File: src/machine/board_nicenano.go (hunk 1 above)
Comment (premek) [reply src/machine/board_nicenano.go:83]: that's what I can see in dmesg output, is that correct?
```
[72606.921963] usb 1-4: New USB device found, idVendor=239a, idProduct=0029, bcdDevice= 1.00
[72606.921967] usb 1-4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[72606.921969] usb 1-4: Product: nice!nano
[72606.921971] usb 1-4: Manufacturer: Nice Keyboards
```

**Comment by premek** [comment]
Hi, thanks for having a look at my PR,

> Regarding the UART, I2C, and SPI, since the nRF52840 is very flexible on assigning IO, you could probably use whichever pins are broken out on the board and not used by anything else. The pinout/schematic does appear to show what is used/available https://github.com/Nicell/nice-keyboards-docs/blob/master/docs/nice!nano/pinout_schematic.md


Sorry, I still don't quite understand this, does it mean I should use any of the other pin numbers like for example:
```
	UART_RX_PIN = P0_06
￼	UART_TX_PIN = P0_08
￼	SDA_PIN = P0_17
￼	SCL_PIN = P0_20
￼	SPI0_SCK_PIN = P0_22
￼	SPI0_SDO_PIN = P0_24
￼	SPI0_SDI_PIN = P0_24
```

File: src/machine/board_nicenano.go (hunk 1 above)
Comment (deadprogram) [reply src/machine/board_nicenano.go:83]: Yes, that is correct.

**Comment by deadprogram** [comment]
```
	UART_RX_PIN = P0_06
￼	UART_TX_PIN = P0_08
￼	SDA_PIN = P0_17
￼	SCL_PIN = P0_20
￼	SPI0_SCK_PIN = P0_22
￼	SPI0_SDO_PIN = P0_24
￼	SPI0_SDI_PIN = P0_24
```

with the exception of needing to change `SPI0_SDI_PIN = P0_24` to `SPI0_SDI_PIN = P1_00` or something, that should work? You would need to confirm that on your actual board.

**Comment by premek** [comment]
I can verify with the board, but the problem is I don't know enough about SPI to test it. I guess just verifying that blinky compiles & runs is not enough?

**Comment by premek** [comment]
I rebased on current `dev` and did the proposed changes in pin numbers. Sorry the `P0_24` was a typo even in the original `D100`. I checked all the numbers again.

**Comment by deadprogram** [comment]
Thanks for making the changes @premek seems alright. I will assume that it is working for you now as expected.

Now merging, thank you again for the contribution.

**Pull request merged**